src/Helper.ts

```diff
@@ -18,11 +18,22 @@ export class Helper {
       const loaded = page.waitForNavigation({
         waitUntil: "load",
       });
-
-      await page.setContent(html);
+      if(url) {
+        await page.goto(url, { waitUntil: ['domcontentloaded', 'networkidle0', 'load'] });
+      }
+      console.log(html)
+      if(html) {
+        await page.setContent(html, { waitUntil: ['domcontentloaded', 'networkidle0', 'load'] });
```

Comment:
@allison-strandberg the `data:text/html,${html}` solution was a mirage. it took me hours to debug that it was failing because it was not handling encodings and special characters properly, and encodeURIComponent did not help. turns out puppeteer now supports `waitUntil` on `setContent` anyway.